### DISCUSSION TITLE
Is there a way to replace text in an image

### DISCUSSION THREAD

**Discussion by mediotex** [ImageMagick/ImageMagick#6549] [Help] [answered]
### ImageMagick version

6.9.11-60

### Operating system, version and so on

Debian 11

### Description

Is there a way to replace text in an image without resampling and degrading quality? I just want to make it crisp and different color.

**Comment by snibgo** [comment]
If the image is raster, then the text is simply pixels of a certain colour or colours. You could remove the text by making it transparent and then filling the holes (aka "inpainting"), and then re-write the text.

Resampling should not be needed.

**Reply by mediotex** [reply]
Yes, raster image.  What should be command for this?

**Comment by snibgo** [comment, marked as answer]
This image has the text written over a fairly constant nearly black background. We can paint that out, then write new text there. Windows BAT syntax, where `humanize.png` is your input image.
```
magick ^
  humanize.png ^
  -fill sRGB(2%%,2%%,2.3%%) ^
  -draw "rectangle 230,1808 2270,2260" ^
  -fill White ^
  -pointsize 400 ^
  -gravity South ^
  -annotate +0+200 "Hello world" ^
  x.png
```

**Comment by mediotex** [comment]
Thank you for the tips. But how to make two lines of text, with different font size and font type? I tried
`annotate +0+200 "text 0,150 'First line of text\nSecond line'" `
but doesn't work as expected.

**Comment by snibgo** [comment]
You can have two or more `-annotate` operations. Before each one, use whatever `-pointsize` and `-font` you want.

**Comment by mediotex** [comment]
got errors `convert-im6.q16: unable to open image `annotate': No such file or directory @ error/blob.c/OpenBlob/2924.
`

**Comment by snibgo** [comment]
Please show your command. `-annotate` needs a hyphen.

**Comment by mediotex** [comment]
`convert gastaldi.png -fill sRGB\(2%%,2%%,2.3%%\) -draw "rectangle 230,1808 2270,2260" -fill White -font 'Arial' -pointsize 200 -gravity South -annotate +0+200 "The first line"  -font 'URW Gothic' -pointsize 300 -gravity South -annotate +0+200 "The Second line" out.png`
plus error:
_convert-im6.q16: unable to read font `URW Gothic' @ warning/annotate.c/RenderType/949._

**Comment by snibgo** [comment]
You use the same offsets "`+0+200`" for both annotations. I suggest using different offsets, eg set one to "`+0+100`".

"`magick -list font`" will list the available fonts.

**Comment by mediotex** [comment]
so this works:
`convert picture.png -fill sRGB\(2%%,2%%,2.3%%\) -draw "rectangle 230,1808 2270,2260" -fill White -font 'Helvetica-Bold' -pointsize 250 -gravity South -annotate +0+200 "THE 2ND LINE" -font 'URW-Gothic-Book' -pointsize 85 -gravity South -annotate +0+570 "THE FIRST LINE" out.png`
